Add triggering_user_name to DagRunProtocol interface

dheerajturaga opened the pull request:
Enable tasks running in isolated environments to access the username
  that triggered a DAG run without requiring direct database access.

  Changes:
  - Add triggering_user_name field to DagRunProtocol interface
  - Update task-sdk DagRun data model to include triggering_user_name
  - Update execution API DagRun model to include triggering_user_name
  - Ensure field flows from database through API to task context

  This allows developers to identify DAG run triggers from within task
  code using the existing DagRunProtocol interface.

```python
def find_owner(dag_run: DagRunProtocol) -> str:
    """
    Extract the owner information from the DAG run.

    Args:
        dag_run: The DAG run protocol object containing run information

    Returns:
        str: Information about who triggered the DAG run
    """
    triggering_user = dag_run.triggering_user_name

    if triggering_user:
        return f"DAG was triggered by user: {triggering_user}"
    else:
        return "DAG was triggered automatically (no user specified)"
```

dheerajturaga committed:
Add triggering_user_name to DagRunProtocol interface

  Enable tasks running in isolated environments to access the username
  that triggered a DAG run without requiring direct database access.

  Changes:
  - Add triggering_user_name field to DagRunProtocol interface
  - Update task-sdk DagRun data model to include triggering_user_name
  - Update execution API DagRun model to include triggering_user_name
  - Ensure field flows from database through API to task context

  This allows developers to identify DAG run triggers from within task
  code using the existing DagRunProtocol interface.

dheerajturaga committed:
Fix tests!

jscheffl reviewed (approved):
Looks reasonable for me.

dheerajturaga committed:
Add Cadwyn migration for triggering_user_name field in DagRunProtocol

  Add backward compatibility migration for the new triggering_user_name field
  added to the DagRun model in the Execution API. This ensures older API
  clients continue to work seamlessly with newer API servers.

dheerajturaga committed:
Add tests for older versions where triggering user is missing

dheerajturaga committed:
Remove triggering user from v2025-04-28

kaxil commented on airflow-core/tests/unit/api_fastapi/execution_api/versions/head/test_task_instances.py:
```diff
@@ -189,6 +189,7 @@ def test_ti_run_state_to_running(
                 "end_date": None,
                 "run_type": "manual",
                 "conf": {},
+                "triggering_user_name": None,
```
Worth adding another unit test that has a field set than None

kaxil reviewed (approved):
1 test suggestion otherwise lgtm

dheerajturaga committed:
Add a test with triggering_user_name set
